File: backend/app/models/document.py
from sqlalchemy import Column, String, Integer, Text, DateTime, JSON, ForeignKey
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import relationship
from datetime import datetime
import uuid

from . import Base

class Document(Base):
    __tablename__ = "documents"
    
    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    filename = Column(String(255), nullable=False)
    content_type = Column(String(100), nullable=False)
    file_size = Column(Integer, nullable=False)
    content = Column(Text)
    
    # Processing
    status = Column(String(50), default="pending")  # pending, processing, completed, failed
    error_message = Column(Text)
    
    # Metadata
    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)
    metadata = Column(JSON, default={})
    
    # Relationships
    collection_id = Column(UUID(as_uuid=True), ForeignKey("collections.id"))
    collection = relationship("Collection", back_populates="documents")

File: frontend/src/types/document.ts
export interface Document {
  id: string;
  filename: string;
  content_type: string;
  file_size: number;
  status: 'pending' | 'processing' | 'completed' | 'failed';
  created_at: string;
  updated_at: string;
  collection_id?: string;
}

export interface SearchResult {
  id: string;
  score: number;
  document_id: string;
  text: string;
  chunk_index: number;
  timestamp?: string;
}

export interface SystemStatus {
  status: string;
  version: string;
  mode: string;
  services: Record<string, string>;
  configuration: {
    max_file_size_mb: number;
    chunk_size: number;
    embedding_model?: string;
  };
}